pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -284,11 +284,6 @@ func (pl *dynamicResources) PreEnqueue(ctx context.Context, pod *v1.Pod) (status
 // pod schedulable. It errs on the side of letting a pod scheduling attempt
 // happen.
 func (pl *dynamicResources) isSchedulableAfterClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
-	if newObj == nil {
-		// Deletes don't make a pod schedulable.
-		return framework.QueueSkip
-	}
```

Comment:
I agree that this is probably never reached, but I am on the edge whether the check should be removed outright.

If we never expect this to happen, should an error be logged when it occurs? Should the code panic?

If the check gets removed for performance reasons, then at least add a comment to the function that explains that `newObj` will never be nil and why.

Reply:
Given that this will never happen, if an error occurs because of the removal of this code, then there is a problem with the code elsewhere and the problem should be exposed, not skip here, what do you think.

And i agree with adding a comment explains that `newObj` will never be nil and why

Reply:
Okay, let's replace with a comment.

Reply:
I have updated the comment, please review